plugin/trino-memsql/src/main/java/io/trino/plugin/memsql/MemSqlClientModule.java

```diff
@@ -55,8 +57,14 @@ public static ConnectionFactory createConnectionFactory(BaseJdbcConfig config, C
 
         return new DriverConnectionFactory(
                 new Driver(),
-                config.getConnectionUrl(),
+                ensureUrlBackwardCompatibility(config.getConnectionUrl()),
```

Comment:
I noticed our config is mutable: so while `config.getConnectionUrl()` would still return "jdbc:mariadb", it would be possible to call `config.setConnectionUrl(...)`, but this seems like it would cause impossible to diagnose issues (e.g., depending on when the Module is created, other components may have already "taken a copy" of the original url).

Should we create an `ImmutableConfig` wrapper around the config (override all `.set` methods w/`throws UnsupportedOperationException("config is immutable")`) to make sure it isn't changing after a certain point in the startup lifecycle?